Implementing Video Conferencing with ReactJS, NodeJS, and WebRTC (#3087)
section-engineering-education/engineering-education: content/articles/implementing-video-conferencing-with-react-node/index.md

- Server Errors: these errors occur in the server and prevents appropriate server response to client-side queries and mutations.
- Transaction Errors: these errors occur while a server update such as a mutation is being executed
- Apollo Client Errors: these errors occur within the core or corresponding libraries.

To set error policy on each request the following code block provided by Graphql should be integrated. Any error reported will fall under the ‘error’ prop along side the data returned from the server or cache.

```javascript
    const MY_QUERY = gql`
      query WillFail {
        badField
        goodField
      }
    `;
    function ShowingSomeErrors() {
      const { loading, error, data } = useQuery(MY_QUERY, { errorPolicy: 'all' });

      if (loading) return <span>loading...</span>
      return (
        <div>
          <h2>Good: {data.goodField}</h2>
          <pre>Bad: {error.graphQLErrors.map(({ message }, i) => (
            <span key={i}>{message}</span>
          ))}
          </pre>
        </div>
      );
    }
```

#### Ignoring errors

> Errors may be ignored during application development. These errors are expected errors that the developer feels unbothered about, to achieve that the following code block should be integrated.

Reviewer: ```suggestion
Errors may be ignored during application development. These errors are expected by the developer and feels unbothered about them.  To achieve that the following code block should be integrated:
```
Explain your point here clearly.

```javascript
    onError(({ response, operation }) => {
      if (operation.operationName === "IgnoreErrorsQuery") {
        response.errors = null;
      }
    })
```

#### Using React testing library

> React testing library is an excellent tool used in application modules and components testing. It is a lightweight solution for testing react components as it provides utility functions on top of **react-dom** and **react-dom/test-utils**. This is implemented using the following command:

Reviewer: ```suggestion
React testing library is an excellent tool used in application modules and components testing. It is a lightweight solution for testing react components. It provides utility functions on top of `react-dom` and `react-dom/test-utils`. We implement it using the following command:
```
